File: easy/392. Is Subsequence/392. Is Subsequence.ts
function isSubsequence(s: string, t: string): boolean {
    if (!s) return true
    if (!t) return false

    let s_i = 0

    for (const symbol of t) {
        if (s[s_i] == symbol)
            s_i += 1

        if (s_i == s.length)
            return true
    }

    return false
};

File: easy/392. Is Subsequence/392. Is Subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if not s or not t:
            return True

        s_index = 0

        for symbol in t:
            if s[s_index] == symbol:
                s_index += 1

        return len(s) == s_index


solution = Solution()
print(solution.isSubsequence("abc", "ahbgdc"))
print(solution.isSubsequence("axc", "ahbgdc"))
